# This is the a_bot specific controller file defining the prefixed joints

# Settings for ros_control control loop
hardware_control_loop:
   loop_hz: 500

# Settings for ros_control hardware interface
ur_hardware_interface:
   joints: &robot_joints
     - shoulder_pan_joint
     - shoulder_lift_joint
     - elbow_joint
     - wrist_1_joint
     - wrist_2_joint
     - wrist_3_joint

# Publish all joint states ----------------------------------
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: 500

# Publish wrench ----------------------------------
force_torque_sensor_controller:
   type:         force_torque_sensor_controller/ForceTorqueSensorController
   publish_rate: 500

# Publish speed_scaling factor
speed_scaling_state_controller:
   type:         scaled_controllers/SpeedScalingStateController
   publish_rate: 500

# Joint Trajectory Controller - position based -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
scaled_pos_joint_traj_controller:
   type: position_controllers/ScaledJointTrajectoryController
   joints: *robot_joints
   constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_pan_joint: {trajectory: 0.2, goal: 0.1}
      shoulder_lift_joint: {trajectory: 0.2, goal: 0.1}
      elbow_joint: {trajectory: 0.2, goal: 0.1}
      wrist_1_joint: {trajectory: 0.2, goal: 0.1}
      wrist_2_joint: {trajectory: 0.2, goal: 0.1}
      wrist_3_joint: {trajectory: 0.2, goal: 0.1}
   stop_trajectory_duration: 0.5
   state_publish_rate: 500
   action_monitor_rate: 10

pos_joint_traj_controller:
   type: position_controllers/JointTrajectoryController
   joints: *robot_joints
   constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_pan_joint: {trajectory: 0.2, goal: 0.1}
      shoulder_lift_joint: {trajectory: 0.2, goal: 0.1}
      elbow_joint: {trajectory: 0.2, goal: 0.1}
      wrist_1_joint: {trajectory: 0.2, goal: 0.1}
      wrist_2_joint: {trajectory: 0.2, goal: 0.1}
      wrist_3_joint: {trajectory: 0.2, goal: 0.1}
   stop_trajectory_duration: 0.5
   state_publish_rate: 500
   action_monitor_rate: 10

cartesian_compliance_controller:
   type: "position_controllers/CartesianComplianceController"
   end_effector_link: "tool0"
   robot_base_link: "base_link"
   ft_sensor_ref_link: "wrist_3_link"
   compliance_ref_link: "tool0"
   target_frame_topic: "target_frame"
   joints:
   - shoulder_pan_joint
   - shoulder_lift_joint
   - elbow_joint
   - wrist_1_joint
   - wrist_2_joint
   - wrist_3_joint

   stiffness: # w.r.t. compliance_ref_link
      trans_x: 500
      trans_y: 500
      trans_z: 500
      rot_x: 20
      rot_y: 20
      rot_z: 20
      
      use_parallel_force_position_control: True

      sel_x: 1.0 
      sel_y: 1.0
      sel_z: 1.0 
      sel_ax: 1.0
      sel_ay: 1.0
      sel_az: 1.0
     
   solver:
      error_scale: 0.5
      iterations: 1
   
   pd_gains:
      trans_x: {p: 0.05}
      trans_y: {p: 0.05}
      trans_z: {p: 0.05}
      rot_x: {p: 0.01}
      rot_y: {p: 0.01}
      rot_z: {p: 0.01}

   gravity:
      x: 0
      y: 0
      z: 0

   tool:
      # Mass of everything that's mounted after the sensor.
      mass: 0

      # The center of mass in the ft_sensor_ref_link.
      # Take a measure and try to estimate that as good as possible.
      com_x: 0.0
      com_y: 0.0
      com_z: 0.0

motion_control_handle:
   type: "cartesian_controllers/MotionControlHandle"
   end_effector_link: "tool0"
   robot_base_link: "base_link"
   target_frame_topic: "target_frame"
   joints:
   - shoulder_pan_joint
   - shoulder_lift_joint
   - elbow_joint
   - wrist_1_joint
   - wrist_2_joint
   - wrist_3_joint